scripts/build-json/build-page-json.js

```diff
@@ -116,35 +39,51 @@ async function buildPageJSON(docsPath) {
 
     // check whether this is a buildable item
     if (data && data.recipe) {
-        // build the item
-        let item = null;
-
+        let body = null;
+        let relatedContentSpec = data.related_content;
+        let contributors = null;
         let elementDirectory = docsDirectory;
 
         switch (data.recipe) {
             case 'guide':
-                item = await guide.buildGuidePageJSON(docsDirectory, data, content);
+                body = await guidePage.buildGuideContentJSON(docsDirectory, data, content);
                 // Guide pages are special because they don't have their own 
                 // directory. Instead, individual .md files share a directory. 
                 // So we need to override the name of the directory to write to.
                 elementDirectory = path.join(docsDirectory, path.basename(docsPath).split('.')[0]);
                 break;
             case 'landing-page':
-                item = await landing.buildLandingPageJSON(docsDirectory, data, content);
+                body = await landingPage.buildLandingPageJSON(docsDirectory, data, content);
                 // Landing pages are special because they don't have their own
                 // directory. Instead, individual .md files share a directory.
                 // So we need to override the name of the directory to write to.
                 elementDirectory = path.join(docsDirectory, path.basename(docsPath).split('.')[0]);
                 break;
-            case 'html-element':
-                item = await buildFromRecipe(docsDirectory, data, content);
-                break;
+            case 'html-element': {
+                    // for recipe-driven content, related_content is in the recipe
+                    const recipePath = path.join(__dirname, '..', '..', 'recipes', `${data.recipe}.yaml`);
+                    const recipe = yaml.safeLoad(fs.readFileSync(recipePath, 'utf8'));
+                    relatedContentSpec = recipe.related_content;
```

Comment:
I know it's going to have to change that we can't have a switch case for every single thing that isn't guide page or landing page. 
Either way, it would be nice to avoid the disk IO and "cache" the `recipe` object based on the `data.recipe` input. We can tackle that some other day.

Reply:
> I know it's going to have to change that we can't have a switch case for every single thing that isn't guide page or landing page.

This has too many negatives for me to understand, but: I hope we will have just one case for all recipe-based pages. But you're right of course, we should cache the recipes here.